dimod/bqm/adjarraybqm.pyx

```diff
@@ -120,30 +120,40 @@ cdef class cyAdjArrayBQM:
         self._idx_to_label = dict()
 
 
-    def __init__(self, object arg1=0):
+    def __init__(self, object obj=0, object vartype=None):
+
+        # handle the case where only vartype is given
+        if vartype is None:
+            try:
+                vartype = obj.vartype
+            except AttributeError:
+                vartype = obj
+                obj = 0
+        self.vartype = as_vartype(vartype)
         
         cdef Bias [:, :] D  # in case it's dense
         cdef size_t num_variables, num_interactions, degree
         cdef VarIndex ui, vi
         cdef Bias b
         cdef cyAdjArrayBQM other
 
-        if isinstance(arg1, Integral):
-            self.invars_.resize(arg1)
-        elif isinstance(arg1, tuple):
-            self.__init__(cyAdjMapBQM(arg1))  # via the map version
-        elif hasattr(arg1, "to_adjarray"):
+        if isinstance(obj, Integral):
+            self.invars_.resize(obj)
+        elif isinstance(obj, tuple):
+            self.__init__(cyAdjMapBQM(obj, vartype))  # via the map version
```

Comment:
This might work in the current implementation, but I don't see why you would generally want to do this.

Reply:
Anecdotally it seems faster to cast it to the map/vector version of cyBQM then copy (in C) to the array version rather than try to figure out the sizes/degrees/labels needed in python space. Needs more testing though